aep/0136.yaml

```diff
@@ -0,0 +1,65 @@
+aliases:
+  CustomOperation:
+    description: A custom method is identified by a path containing a colon (:) before the verb
+    targets:
+      - formats: ['oas2', 'oas3']
+        given:
+          # condition excludes custom methods Update Array custom operations defined in AEP-144, as different list of rules applies to them
+          - $.paths[?(@property.match(/:(?!add|remove)([^/]*)$/))]
+
+rules:
+  aep-136-http-method:
+    description: Custom methods should use POST or GET.
+    severity: warn
+    formats: ['oas2', 'oas3']
+    given: '#CustomOperation'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          properties:
+            post: {}
+            get: {}
+          additionalProperties: false
```

Comment:
`additionalProperties: false` is also problematic here -- it will cause a false negative on any path that has `parameters` or specification extensions. This one is less clear how to fix. We could go the same route as below -- not / anyOf / required -- and name the other methods not allowed, which I guess would be `put`,`options`, `head`, and `trace`. But now I wonder if by their exclusion these are actually errors rather than warnings. Might be a good topic for the meeting tomorrow.

Reply:
FYI, we did discuss this in last week's AEP meeting and decided that we only need one rule, which will issue a warning for any custom method that does not use GET or POST.